components/site/src/link_checking.rs

```diff
@@ -148,24 +166,61 @@ pub fn check_external_links(site: &Site) -> Result<()> {
             if should_skip_by_prefix(&external_link, &site.config.link_checker.skip_prefixes) {
                 skipped_link_count += 1;
             } else {
-                let domain = get_link_domain(&external_link)?;
+                let domain = get_link_domain(&external_link);
                 checked_links.push(LinkDef::new(s.file.path.clone(), external_link, domain));
             }
         }
     }
 
+    // separate the links with valid domains from the links with invalid domains
+    let (checked_links, invalid_url_links): (Vec<&LinkDef>, Vec<&LinkDef>) =
+        checked_links.iter().partition(|link| link.domain.is_ok());
+
+    // get any domains that failed to parse and log them at the configured log level
+    let invalid_link_errs: Vec<String> = invalid_url_links
+        .iter()
+        .map(|link: &&LinkDef| link.domain.as_ref().unwrap_err().clone())
+        .collect();
+
     println!(
-        "Checking {} external link(s).  Skipping {} external link(s).",
+        "Checking {} external link(s). Skipping {} external link(s).{}",
         checked_links.len(),
-        skipped_link_count
+        skipped_link_count,
+        if invalid_link_errs.is_empty() {
+            "".to_string()
+        } else {
+            format!(" {} link(s) had unparseable URLs.", invalid_link_errs.len())
+        }
     );
 
+    if !invalid_link_errs.is_empty() {
+        match site.config.link_checker.external_level {
+            // panic if the link checker level is set to error.  bail! can only take one error
+            // message, and it prefixes "Error: ", but we may have accumulated many errors, and it
+            // loos weird if only the first line says "Error: ", so we use join() here to add the
+            // ErrorLevel's log_prefix to each line.
```

Comment:
@Keats here's a description of the discrepancy between prefixing on errors and warnings.